Keypad/Keypad.js

```diff
@@ -0,0 +1,327 @@
+		 *
+		 * @type {Function}
+		 * @param {Object} event
+		 * @public
+		 */
+		onKeyButtonClick: PropTypes.func,
+
+		/**
+		 * Text displayed below the number/icon/symbol on the key.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		subtext: PropTypes.string,
+
+		/**
+		 * Text displayed in the center of the key.
+		 *
+		 * @type {String}
+		 * @public
+		 */
+		text: PropTypes.string
+	},
+
+	styles: {
+		css,
+		className: 'key'
+	},
+
+	computed: {
+		textComponent: ({text}) => {
+			return (text != null && text !== '') ? <span className={css.text}>{text}</span> : null;
+		},
+		subtextComponent: ({subtext}) => {
+			return (subtext != null && subtext !== '') ? <span className={css.subtext}>{subtext}</span> : null;
+		}
+	},
+
+	handlers: {
+		onClick: handle(
+			forward('onClick'),
+			adaptEvent((ev, {children: key}) => ({key}), forward('onKeyButtonClick'))
+		)
+	},
+
+	render: ({children, subtextComponent, textComponent, ...rest}) => {
+		delete rest.onKeyButtonClick;
+
+		return (
+			<div className={css.keyContainer}>
+				<Button
+					{...rest}
+					css={css}
+					icon={children}
+					size="large"
+				>
+					{textComponent}
+					{subtextComponent}
+				</Button>
+			</div>
+		);
+	}
+});
+
+/**
+ * The basic Layout of a Keypad component.
+ *
+ * @class KeypadBase
+ * @memberof agate/Keypad
+ * @extends agate/Keypad.Key
+ * @ui
+ * @public
+ */
+const KeypadBase = kind({
+	name: 'Key',
+
+	propTypes: /** @lends agate/Keypad.KeypadBase.prototype */{
+		/**
+		 * Applies a disabled style and the control becomes non-interactive.
+		 *
+		 * @type {Boolean}
+		 * @default false
+		 * @public
+		 */
+		disabled: PropTypes.bool,
+
+		/**
+		 * Called when a button is clicked. Includes the 'key' key in its event payload, updates the state and the input value accordingly.
+		 *
+		 * @type {Function}
+		 * @param {Object} event
+		 * @public
+		 */
+		handleInputValue: PropTypes.func
+	},
+
+	styles: {
+		css,
+		className: 'keypad'
+	},
+
+	render: ({disabled, handleInputValue, ...rest}) => {
+		return (
+			<Layout {...rest} align="center end" className={css.keypad} inline wrap>
+				{KEY_LIST.map((keyText, rowIndex) => {
+					return (
+						<Cell
+							aria-label={keyText.text === 'arrowleftturn' ? $L('Back Space') : keyText.text}
+							component={Key}
+							disabled={disabled}
+							key={`key${rowIndex}-${keyText.text}`}
+							onKeyButtonClick={() => handleInputValue(keyText.text)}
+							shrink
+							subtext={keyText.subtext}
+							text={keyText.text === 'arrowleftturn' || keyText.text === 'phone' ? null : keyText.text}
+						>
+							{keyText.text === 'arrowleftturn' || keyText.text === 'phone' ? keyText.text : null}
+						</Cell>
+					);
+				})}
+			</Layout>
+		);
+	}
+});
+
+/**
+ * A Keypad component with an Input to display the outcome.
+ *
+ * @class Keypad
+ * @memberof agate/Keypad
+ * @extends agate/Keypad.KeypadBase
+ * @ui
+ * @public
+ */
+class Keypad extends React.Component {
```

Comment:
If this was converted to a HigherOrderComponent (HOC) it would be more conveniently compatible with the Skinning HOC and follow the construction convention better. If it needs to remain as _not_ a HOC, then KeypadBase should be wrapped in Skinnable before it's used by this Component. _In any case, Keypad, somewhere in the line, needs to be wrapped in Skinnable.

Reply:
`Keypad` is now converted to `KeypadExtended` HOC. And now `KeypadBase` is wrapped in `Skinnable`.